src/exchanges/gdax/GDAXExchangeAPI.ts

```diff
@@ -92,9 +92,10 @@ export class GDAXExchangeAPI implements PublicExchangeAPI, AuthenticatedExchange
     loadMidMarketPrice(product: string): Promise<BigJS> {
         return this.loadTicker(product).then((ticker) => {
             if (!ticker || !ticker.bid || !ticker.ask) {
-                throw new HTTPError(`Loading midmarket price for ${product} failed because ticker data was incomplete or unavailable`, {status: 200, body: ticker});
+                const err = new HTTPError(`Loading midmarket price for ${product} failed because ticker data was incomplete or unavailable`, {status: 200, body: ticker});
+                return Promise.reject(err);
             }
-            return ticker.ask.plus(ticker.bid).times(0.5);
+            return Promise.resolve(ticker.ask.plus(ticker.bid).times(0.5));
```

Comment:
This is in a `then`

Reply:
@fb55 This is actually required because the if block changed from throwing an Error to returning Promise.reject(). It appears that in general, tsc needs an explicit Promise.resolve() in any function/method where Promise.reject() is needed:

```
src/exchanges/gdax/GDAXExchangeAPI.ts(94,46): error TS2345: Argument of type '(ticker: Ticker) => BigNumber | Promise<never>' is not assignable to parameter of type '(value: Ticker) => PromiseLike<never>'.
  Type 'BigNumber | Promise<never>' is not assignable to type 'PromiseLike<never>'.
    Type 'BigNumber' is not assignable to type 'PromiseLike<never>'.
      Property 'then' is missing in type 'BigNumber'.
22:15:54 - Compilation complete. Watching for file changes.
```